helm-chart/templates/secrets.yaml

```diff
@@ -0,0 +1,106 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: local-settings-{{ .Values.deployEnv }}
+  labels:
+    deployEnv: "{{ .Values.deployEnv }}"
+type: Opaque
+data:
+  local_settings.py: {{ .Files.Get "local_settings.py" | b64enc }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: google-client-secret-{{ .Values.deployEnv }}
+  labels:
+    deployEnv: "{{ .Values.deployEnv }}"
+type: Opaque
+data:
+  client_secrets.json: {{ .Files.Get "secrets/client_secrets.json" | b64enc }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: varnish-secret-{{ .Values.deployEnv }}
+  labels:
+    deployEnv: "{{ .Values.deployEnv }}"
+type: Opaque
+data:
+  varnish-secret: {{ .Files.Get "secrets/varnish-secret" | b64enc }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: school-lookup-data-{{ .Values.deployEnv }}
+  labels:
+    deployEnv: "{{ .Values.deployEnv }}"
+type: Opaque
+data:
+  schools.tsv: {{ .Files.Get "secrets/schools.tsv" | b64enc }}    
+---
+apiVersion: v1
+kind: Secret
+metadata: 
+  name: backup-manager-secret-{{ .Values.deployEnv }}
+  labels:
+    deployEnv: "{{ .Values.deployEnv }}"
+type: Opaque
+data:
+  BackupManagerKey.json: {{ .Files.Get "secrets/BackupManagerKey.json" | b64enc }}
```

Comment:
the filenames should be variables in the values.yml

Reply:
Do we want it to be a file reference variable? The solution we discussed previously was to provide a ref to an existing secret, with an option to pass in data to create a new secret. That might not align with the proposal to manage cauldrons via gitops, since there isn't really a way to encrypt the values section of a helmRelease that is stored in repo. A file reference var definitely won't work with a helmRelease

Reply:
👍